pkg/reconcilermanager/controllers/reposync_controller.go

```diff
@@ -170,7 +171,12 @@ func (r *RepoSyncReconciler) Reconcile(ctx context.Context, req controllerruntim
 		return controllerruntime.Result{}, updateErr
 	}
 
-	if err := r.setup(ctx, reconcilerRef, rsRef, currentRS, rs); err != nil {
+	err := r.setupOrTeardown(ctx, rs, func(ctx context.Context) error {
+		return r.setup(ctx, reconcilerRef, rsRef, currentRS, rs)
+	}, func(ctx context.Context) error {
+		return r.teardown(ctx, reconcilerRef, rsRef, currentRS, rs)
+	})
```

Comment:
Can you define two variables for the last two parameters of `setupOrTeardown`? 

```
var setupFn = ...
var teardownFn = ...
err := r.setupOrTeardown(ctx, rs, setupFn, teardownFn)
```

Reply:
done